Yosemite/Yosemite/Stores/ShippingLabelStore.swift

```diff
@@ -212,6 +220,49 @@ private extension ShippingLabelStore {
             }
         }
     }
+
+    func purchaseShippingLabel(siteID: Int64,
+                               orderID: Int64,
+                               originAddress: ShippingLabelAddress,
+                               destinationAddress: ShippingLabelAddress,
+                               packages: [ShippingLabelPackageSelected],
+                               emailCustomerReceipt: Bool,
+                               completion: @escaping (Result<[ShippingLabel], Error>) -> Void) {
+        var labelPurchaseIDs: [Int64] = []
+
+        // Make the initial purchase request.
+        remote.purchaseShippingLabel(siteID: siteID,
+                                     orderID: orderID,
+                                     originAddress: originAddress,
+                                     destinationAddress: destinationAddress,
+                                     packages: packages,
+                                     emailCustomerReceipt: emailCustomerReceipt) { result in
+            switch result {
+            case .success(let labelPurchases):
+                // Save the IDs of label purchases in the response so we can poll the backend for their status.
+                for label in labelPurchases {
+                    labelPurchaseIDs.append(label.shippingLabelID)
+                }
+            case .failure(let error):
+                DDLogError("⛔️ Error purchasing shipping label for order \(orderID): \(error)")
+                completion(.failure(error))
+            }
+        }
+
+        // Wait for 2 seconds (give the backend time to process the purchase)
+        DispatchQueue.main.asyncAfter(deadline: .now() + 2.0) { [weak self] in
+            guard let self = self else { return }
+
+            // Poll the status of the label purchases from the response above
+            // with a delay of 1 second each time, with a maximum of 3 retries
+            self.pollLabelStatus(withDelayInSeconds: 1.0,
+                                 maxRetries: 3,
+                                 siteID: siteID,
+                                 orderID: orderID,
+                                 labelIDs: labelPurchaseIDs,
+                                 completion: completion)
+        }
```

Comment:
Should we do this just in case of success? Looking at the flow chart, yes.

I don't think it makes sense to start polling, regardless of the result of the previous call. Furthermore, we could happen in the event that the first call for purchase becomes very slow, that the `pollLabelStatus` responds sooner.
And if the purchase API request fails, we would still make an unnecessary call. I'm wrong?

Reply:
Oh yes, thanks for catching that! I was mistaken about the flow when I wrote it that way. That polling should be only in the success case; I've changed that in f598fae.

Reply:
5 (2+1+1+1) seconds seems short to me - my experience buying labels through wp-admin (on shared hosting, admittedly) seems at least that long, if not more. cc @DanReyLop

Reply:
The idea here was to match the web client's polling behavior (more discussion in p91TBi-4K3-p2). The delay should be between when we get the response and when we next poll, so the total time could be longer than 5 seconds. If I've misunderstood any of that, or if we think we need to lengthen the delay, I can certainly change this!

Reply:
5 seconds is definitely too low. But, as Rachel pointed out, when including 3 `iOS -> WP -> WCS server` round-trips, the actual time will be significantly higher. It still feels unnecessarily low though, there's no downside to retrying 5 times or so instead of 3.

I see that the logic in the web client [is exactly the same](https://github.com/Automattic/woocommerce-services/blob/dbeed610f336f8db7995743765dbcd0ff2e18474/client/extensions/woocommerce/woocommerce-services/state/shipping-label/actions.js#L757-L777), so whatever is decided is should be applied in both codebases.